doc/SAI-IPv6-Segment-Routing-Update.md

```diff
@@ -0,0 +1,484 @@
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT6,
+
+    /** Endpoint with decapsulation and specific IPv6 table lookup */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT4,
+
+    /** Endpoint with decapsulation and specific IP table lookup */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_DT46,
+
+    /** Endpoint Bound to a policy with Encapsulation */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_ENCAPS,
+
+    /** End.B6.Encaps function with a reduced SRH */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_ENCAPS_RED,
+
+    /** Endpoint Bound to a policy with Insertion */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_INSERT,
+
+    /** End.B6.Insert function with a reduced SRH */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_B6_INSERT_RED,
+
+    /** Custom range base value */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of Custom range base */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_CUSTOM_RANGE_END
+
+} sai_local_sid_entry_endpoint_type_t;
+```
+3. Added enum sai_local_sid_entry_endpoint_flavor_t to list all the combinations of endpoint flavors 
+   as defined in RFC-8986
+
+```
+/**
+ * @brief Enum defining Endpoint Segment flavors for End, End.X and End.T functions
+ */
+typedef enum _sai_local_sid_entry_endpoint_flavor_t
+{
+    /** Penultimate segment pop */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP,
+
+    /** Ultimate Segment pop */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USP,
+
+    /** Ultimate Segment decapsulation */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USD,
+
+    /** PSP and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP,
+
+    /** USD and USP */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_USD_AND_USP,
+
+    /** PSP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USD,
+
+    /** PSP, USP and USD */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP_AND_USP_AND_USD
+
+    /** None */
+    SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_NONE
+
+} sai_local_sid_entry_endpoint_flavor_t;
+```
+
+4. Added a new object sai_local_sid_entry_t configure Endpoint behavior.
+
+```
+/**
+ * @brief Attribute list for Local SID
+ */
+typedef enum _sai_local_sid_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_START,
+
+    /**
+     * @brief Endpoint Function
+     *
+     * @type sai_local_sid_entry_endpoint_type_t
+     * @flags MANDATORY_ON_CREATE | CREATE_AND_SET
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE = SAI_LOCAL_SID_ENTRY_ATTR_START,
+
+    /**
+     * @brief Flavor for End, End.X and End.T functions
+     *
+     * @type sai_local_sid_entry_endpoint_flavor_t
+     * @flags CREATE_AND_SET
+     * @default SAI_LOCAL_SID_ENTRY_ENDPOINT_FLAVOR_PSP
+     * @validonly SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_E or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_X or SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_TYPE == SAI_LOCAL_SID_ENTRY_ENDPOINT_TYPE_T
+     */
+    SAI_LOCAL_SID_ENTRY_ATTR_ENDPOINT_FLAVOR,
```

Comment:
What will be default flavour if this is not specified ? 

Conversely, if were to specify "base" End or EndX behavior, this attr need not set.

Reply:
Default is set to "None" and that will correspond to the "base" behavior. So, if this attribute is not set, base behavior will be assumed.